## Import Necessary Libraries for data handling and manipulation

In [7]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [13]:
# we're gonna save the path in a variable called 'path' - this will allow us to change the file path and perfrom analysis on differernt files
path = "/content/1 to  17 oct in app bb daily.csv"
#use read_csv function to read our csv file
bbdata = pd.read_csv(path)

In [14]:
# display the imported data
bbdata

Attributed Touch Type Attributed Touch Time      Install Time  \
0                impression       9/19/2021 21:39   9/19/2021 21:44   
1                     click      10/17/2021 23:48  10/17/2021 23:48   
2                     click      10/17/2021 23:48  10/17/2021 23:48   
3                     click      10/15/2021 10:40   10/17/2021 9:56   
4                     click      10/17/2021 23:05  10/17/2021 23:05   
...                     ...                   ...               ...   
11794                 click        10/1/2021 0:10    10/1/2021 0:11   
11795                 click       9/18/2021 19:09   9/18/2021 19:10   
11796                 click        10/1/2021 0:00    10/1/2021 0:02   
11797                 click        10/1/2021 0:00    10/1/2021 0:02   
11798                 click       9/30/2021 23:46   9/30/2021 23:47   

             Event Time                Event Name  \
0      10/17/2021 23:59     af_first_subscription   
1      10/17/2021 23:51     af_first_subscription   
2      10/17/2021 23:51            af_address_add   
3      10/17/2021 23:50            af_first_topup   
4      10/17/2021 23:48            af_address_add   
...                 ...                       ...   
11794    10/1/2021 0:15            af_address_add   
11795    10/1/2021 0:12            af_first_topup   
11796    10/1/2021 0:08     af_first_subscription   
11797    10/1/2021 0:08            af_address_add   
11798    10/1/2021 0:00  af_first_wallet_recharge   

                                             Event Value  Event Revenue  \
0      {"af_currency":"INR","af_revenue":30,"af_order...        30.0000   
1      {"af_currency":"INR","af_revenue":29,"af_order...        29.0000   
2      {"af_city":"Noida - Ghaziabad","af_content":"R...            NaN   
3      {"af_currency":"INR","af_content":["Idly Dosa ...       299.7044   
4      {"af_city":"Kolkata","af_content":"Animikha Ho...            NaN   
...                                                  ...            ...   
11794  {"af_city":"Chennai","af_content":"VNR Milford...            NaN   
11795  {"af_currency":"INR","af_content":["Chocolate ...       321.0000   
11796  {"af_currency":"INR","af_revenue":25,"af_order...        25.0000   
11797  {"af_city":"Mumbai","af_content":"Vitthal Plaz...            NaN   
11798              {"af_currency":"INR","af_revenue":20}        20.0000   

      Event Revenue Currency  Event Revenue INR  Cost Model  ...  \
0                        INR            30.0000         NaN  ...   
1                        INR            29.0000         NaN  ...   
2                        USD                NaN         NaN  ...   
3                        INR           299.7044         NaN  ...   
4                        USD                NaN         NaN  ...   
...                      ...                ...         ...  ...   
11794                    USD                NaN         NaN  ...   
11795                    INR           321.0000         NaN  ...   
11796                    INR            25.0000         NaN  ...   
11797                    USD                NaN         NaN  ...   
11798                    INR            20.0000         NaN  ...   

       Is Retargeting  Retargeting Conversion Type Is Primary Attribution  \
0               False                          NaN                   True   
1               False                          NaN                   True   
2               False                          NaN                   True   
3               False                          NaN                   True   
4               False                          NaN                   True   
...               ...                          ...                    ...   
11794           False                          NaN                   True   
11795           False                          NaN                   True   
11796           False                          NaN                   True   
11797           False       

as we can observe, therer are almost 12,000 rows and also 59 unique columns - that is a lot of information. For the case of this problem, we are only going to handle the first 5 columns that inlcudes:
- Attributed Touch Type : wherether the touch type was a click or an impression
- Attributed Touch Time	: the time when the touch happened
- Install Time : Time of installation of app/software
-	Event Time : Time when an event took place
-	Event Name : Name of the event that took place in the above mentioned time

In [15]:
# store only the first five columns in a new variable called 'new_bbdata'
new_bbdata = bbdata.iloc[:,0:5] # use iloc to select the first 5 indexes

In [16]:
# lets check the new variable and see if it has all the necessary columns
new_bbdata

Attributed Touch Type Attributed Touch Time      Install Time  \
0                impression       9/19/2021 21:39   9/19/2021 21:44   
1                     click      10/17/2021 23:48  10/17/2021 23:48   
2                     click      10/17/2021 23:48  10/17/2021 23:48   
3                     click      10/15/2021 10:40   10/17/2021 9:56   
4                     click      10/17/2021 23:05  10/17/2021 23:05   
...                     ...                   ...               ...   
11794                 click        10/1/2021 0:10    10/1/2021 0:11   
11795                 click       9/18/2021 19:09   9/18/2021 19:10   
11796                 click        10/1/2021 0:00    10/1/2021 0:02   
11797                 click        10/1/2021 0:00    10/1/2021 0:02   
11798                 click       9/30/2021 23:46   9/30/2021 23:47   

             Event Time                Event Name  
0      10/17/2021 23:59     af_first_subscription  
1      10/17/2021 23:51     af_first_subscription  
2      10/17/2021 23:51            af_address_add  
3      10/17/2021 23:50            af_first_topup  
4      10/17/2021 23:48            af_address_add  
...                 ...                       ...  
11794    10/1/2021 0:15            af_address_add  
11795    10/1/2021 0:12            af_first_topup  
11796    10/1/2021 0:08     af_first_subscription  
11797    10/1/2021 0:08            af_address_add  
11798    10/1/2021 0:00  af_first_wallet_recharge  

[11799 rows x 5 columns]

## Problem statement

In no particulat order, we want to look at the dataset and gather some insights to help the team at bigbasket make more informed decision to drive their growth and solve key problems that the user face as well as the problem currently occuring at big basket. A good place to start would be the following:

1. Analyze markettting performance in terms of Clicks and impressions
2. We have to analyze events distribution wrt time (pie chart)
3. Event performance analysis wrt time (comparative line chart)
4. And then, We have to analyze installs.

## data pre processing and cleaning

Lets make sure that our dataset does not have any outliers or any missing values. Essentially we want to make sure that we have a clean data before we start working with it.

In [18]:
# use isnull() to turn all misisng / not misisng calues into True or false values. Then use the sum() to get a total number of missing values for each column
new_bbdata.isnull().sum()

Attributed Touch Type    0
Attributed Touch Time    0
Install Time             0
Event Time               0
Event Name               0
dtype: int64

## Simple Descriptive analysis

In [22]:
# lets look at some simple statistics about the dataset usign the descibe function
new_bbdata.describe()

Attributed Touch Type Attributed Touch Time     Install Time  \
count                  11799                 11799            11799   
unique                     2                  5324             5316   
top                    click       4/18/2021 22:24  4/18/2021 22:39   
freq                   11209                    59               59   

             Event Time      Event Name  
count             11799           11799  
unique             7699               4  
top     10/9/2021 13:37  af_address_add  
freq                  9            5664

The First thing we can notice is that there is no mean, median, std deviation, miniumum, maximum or quantiles (25-50-75 percentiles).
Instead we have:
- count : number of entries in each column (this is 11799 for all, since no missing values)
- unique : the number of Unique types of entries in each column.
- top : Indicates the most frequently occurring value within each column.
- frequency : Represents the frequency or count of the top value in each column